blog/2021-05-29-using-rust.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Building a reliable and tRUSTworthy web service 🦀
+author: Rohan
+github: RohanGautam
+description: Explore the use of rust in FoodDX, and go over key architectural decisions
+tags: rust
+---
+
+## Product intro
+
+[FoodDX](https://www.fooddx.com/) is a service which helps people get insights on how to improve their diet with the help from our proprietary AI technology and personalized feedback from our nutrition experts. In it's current stage, it helps score images of food taken through an app, and gives it a score from 1-5, and also provides personalized tips for the food.
+
+It's a big project with a lot of components - the AI models, the app, and the backend infrastructure which handles it all. We'll be taking a look at the backend in this article.
+
+## Lifecycle of an image
+
+Before we dive into the infrastructure, it would help to take a look at what an image goes through once it enters our system.
+
+- The image is assigned a UUID.
+- The image is uploaded to a `s3` bucket.
+- The image is then downloaded, preprocessed, run through our models, and a food tip is generated for the image.
+- The food tip, inference results and the image hash [^1] form the final response.
+
+## Stack decisions: Rust and Haskell
+
+Haskell is used for the client facing API, and it's for other ad-hoc tasks such as reading/writing to a database among others.
+
+Rust is used for image preprocessing, model inference and sending the results back. Sending of results was done differently in the two approaches outlined below. Rust was chosen for it's high efficiency, small executable footprint, and absence of a garbage collector. It also had strong type system, speed & relatively actively maintained Tensorflow (client) library.
+
+# Introduction and pitfalls of the existing architecture
+
+![](../images/blogposts/v2-arch-diagram.png)
```

Comment:
In this image,
<img width="306" alt="Screenshot 2021-05-28 at 11 16 55 AM" src="https://user-images.githubusercontent.com/15181803/119936411-c5445c00-bfa6-11eb-93c7-c8808b7afa3d.png">

I think it should be model is loaded OR models are loaded?

Reply:
ay good catch man! Thanks a lot :D Should def have an "are" in there